core/directory.go

```diff
@@ -647,14 +646,17 @@ func (dir *Directory) WithNewDirectory(ctx context.Context, dest string, permiss
 func (dir *Directory) Diff(ctx context.Context, other *Directory) (*Directory, error) {
 	dir = dir.Clone()
 
-	if dir.Dir != other.Dir {
-		// TODO(vito): work around with llb.Copy shenanigans?
-		return nil, fmt.Errorf("TODO: cannot diff with different relative paths: %q != %q", dir.Dir, other.Dir)
+	thisDirPath := dir.Dir
+	if thisDirPath == "" {
+		thisDirPath = "/"
```

Comment:
Potentially we could use `path.Clean`?

Reply:
https://go.dev/play/p/MZpwcipFm03 I think that would result in comparing "." and "/".

Honestly it would be nice to just make all of this consistent from the start so that `Directory.Dir` is always a consistent value once set, but I'm trying to be minimal here since we want to release this ASAP.